fe/fe-core/src/main/java/com/starrocks/catalog/HudiTable.java

```diff
@@ -0,0 +1,391 @@
+        }
+
+        List<FieldSchema> unPartHiveColumns = metastoreTable.getSd().getCols();
+        List<FieldSchema> partHiveColumns = metastoreTable.getPartitionKeys();
+        for (FieldSchema partHiveColumn : partHiveColumns) {
+            String columnName = partHiveColumn.getName();
+            Column partColumn = this.nameToColumn.get(columnName);
+            if (partColumn == null) {
+                throw new DdlException("partition column [" + columnName + "] must exist in column list");
+            } else {
+                this.partColumnNames.add(columnName);
+            }
+        }
+
+        for (FieldSchema s : unPartHiveColumns) {
+            this.dataColumnNames.add(s.getName());
+        }
+
+        Configuration conf = new Configuration();
+        HoodieTableMetaClient metaClient = HoodieTableMetaClient.builder().setConf(conf).setBasePath(hudiBasePath).build();
+        TableSchemaResolver schemaUtil = new TableSchemaResolver(metaClient);
+        Schema tableSchema;
+        try {
+            tableSchema = HoodieAvroUtils.createHoodieWriteSchema(schemaUtil.getTableAvroSchemaWithoutMetadataFields());
+        } catch (Exception e) {
+            throw new DdlException("Cannot get hudi table schema.");
+        }
+        for (Column column : this.fullSchema) {
+            Schema.Field hudiColumn = tableSchema.getField(column.getName());
+            Schema.Type hudiType = hudiColumn.schema().getTypes().get(1).getType();
+            if (hudiColumn == null) {
+                throw new DdlException("column [" + column.getName() + "] not exists in hudi.");
+            }
+            Set<PrimitiveType> validColumnTypes = getValidColumnType(hudiType);
+            if (!validColumnTypes.contains(column.getPrimitiveType())) {
+                throw new DdlException("can not convert hudi column type [" + hudiType + "] to " +
+                        "starrocks type [" + column.getPrimitiveType() + "], column name: " + column.getName());
+            }
+        }
+
+        if (!copiedProps.isEmpty()) {
+            throw new DdlException("Unknown table properties: " + copiedProps.toString());
+        }
+    }
+
+    private Set<PrimitiveType> getValidColumnType(Schema.Type type) {
+        if (type == null) {
+            return Sets.newHashSet();
+        }
+
+        switch (type) {
+            case BOOLEAN:
+                return Sets.newHashSet(PrimitiveType.BOOLEAN);
+            case INT:
+                return Sets.newHashSet(PrimitiveType.INT, PrimitiveType.TINYINT, PrimitiveType.SMALLINT);
+            case LONG:
+                return Sets.newHashSet(PrimitiveType.BIGINT);
+            case FLOAT:
+                return Sets.newHashSet(PrimitiveType.FLOAT);
+            case DOUBLE:
+                return Sets.newHashSet(PrimitiveType.DOUBLE);
+            case STRING:
+                return Sets.newHashSet(PrimitiveType.VARCHAR, PrimitiveType.CHAR);
```

Comment:
Do we have a typemap between hudi and starrocks?

Reply:
This should be provided in the documentation with hive or iceberg.